File: TransCoder_tokenized_test_set_functions/python/DECODE_STRING_RECURSIVELY_ENCODED_COUNT_FOLLOWED_SUBSTRING.py
def decode ( Str ) :
    integerstack = [ ]
    stringstack = [ ]
    temp = ""
    result = ""
    for i in range ( len ( Str ) ) :
        count = 0
        if ( Str [ i ] >= '0' and Str [ i ] <= '9' ) :
            while ( Str [ i ] >= '0' and Str [ i ] <= '9' ) :
                count = count * 10 + ord ( Str [ i ] ) - ord ( '0' )
                i += 1
            i -= 1
            integerstack . append ( count )
        elif ( Str [ i ] == ']' ) :
            temp = ""
            count = 0
            if ( len ( integerstack ) != 0 ) :
                count = integerstack [ - 1 ]
                integerstack . pop ( )
            while ( len ( stringstack ) != 0 and stringstack [ - 1 ] != '[' ) :
                temp = stringstack [ - 1 ] + temp
                stringstack . pop ( )
            if ( len ( stringstack ) != 0 and stringstack [ - 1 ] == '[' ) :
                stringstack . pop ( )
            for j in range ( count ) :
                result = result + temp
            for j in range ( len ( result ) ) :
                stringstack . append ( result [ j ] )
            result = ""
        elif ( Str [ i ] == '[' ) :
            if ( Str [ i - 1 ] >= '0' and Str [ i - 1 ] <= '9' ) :
                stringstack . append ( Str [ i ] )
            else :
                stringstack . append ( Str [ i ] )
                integerstack . append ( 1 )
        else :
            stringstack . append ( Str [ i ] )
    while len ( stringstack ) != 0 :
        result = stringstack [ - 1 ] + result
        stringstack . pop ( )
    return result

File: results/py/DECODE_STRING_RECURSIVELY_ENCODED_COUNT_FOLLOWED_SUBSTRING.cpp
string f_filled ( string Str ) {
  vector < int > integerstack ;
  vector < string > stringstack ;
  string temp ;
  string result ;
  for ( int i = 0 ;
  i < Str . length ( ) ;
  i ++ ) {
    int count = 0 ;
    if ( ( Str [ i ] >= '0' && Str [ i ] <= '9' ) || ( Str [ i ] >= '0' && Str [ i ] <= '9' ) ) {
      while ( ( Str [ i ] >= '0' && Str [ i ] <= '9' ) || ( Str [ i ] >= '0' && Str [ i ] <= '9' ) ) {
        count = count * 10 + ( Str [ i ] - '0' ) ;
        i ++ ;
      }
      i -- ;
      integerstack . push_back ( count ) ;
    }
    else if ( ( Str [ i ] == ']' ) || ( Str [ i ] == '[' ) ) {
      temp = "" ;
      int count = 0 ;
      if ( ( integerstack . size ( ) != 0 ) && ( integerstack . back ( ) != '[' ) ) {
        count = integerstack . back ( ) ;
        integerstack . pop_back ( ) ;
      }
      while ( ( stringstack . size ( ) != 0 ) && ( stringstack . back ( ) != '[' ) ) {
        temp = stringstack . back ( ) + temp ;
        stringstack . pop_back ( ) ;
      }
      if ( ( stringstack . size ( ) != 0 ) && ( stringstack . back ( ) == '[' ) ) stringstack . pop_back ( ) ;
      for ( int j = 0 ;
      j < count ;
      j ++ ) result = result + temp ;
      for ( int j = 0 ;
      j < result . length ( ) ;
      j ++ ) stringstack . push_back ( result [ j ] ) ;
      result = "" ;
    }
    else if ( ( Str [ i ] == '[' ) ) {
      if ( ( Str [ i - 1 ] >= '0' && Str [ i - 1 ] <= '9' ) || ( Str [ i ] >= '0' && Str [ i - 1 ] <= '9' ) ) stringstack . push_back ( Str [ i ] ) ;
      else {
        stringstack . push_back ( Str [ i ] ) ;
        integerstack . push_back ( 1 ) ;
      }
    }
    else stringstack . push_back ( Str [ i ] ) ;
  }
  while ( stringstack . size ( ) != 0 ) {
    result = stringstack . back ( ) + result ;
    stringstack . pop_back ( ) ;
  }
  return result ;
}
